internal/rbd/sms_controllerserver.go

```diff
@@ -0,0 +1,220 @@
+/*
+Copyright 2025 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rbd
+
+import (
+	"context"
+	"errors"
+
+	rbderrors "github.com/ceph/ceph-csi/internal/rbd/errors"
+	"github.com/ceph/ceph-csi/internal/util/log"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	// defaultMaxResults is the internal limit for max results in metadata streaming
+	// This is set to a reasonable value to prevent excessive memory usage
+	// and ensure efficient processing of metadata requests.
+	// TODO: Consider making this configurable via a cmdline flag.
```

Comment:
Let's have an issue created for this, so it's easier to track.

Reply:
https://github.com/ceph/ceph-csi/issues/5401
created